components/gif_picker/gif_picker.tsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {GfycatAPIItem} from 'types/external/gfycat';
+
+import App from 'components/gif_picker/components/App';
+import Categories from 'components/gif_picker/components/Categories';
+import Search from 'components/gif_picker/components/Search';
+import Trending from 'components/gif_picker/components/Trending';
+import constants from 'components/gif_picker/utils/constants';
+
+export const appProps = {
+    appName: constants.appName.mattermost,
+    basePath: '/mattermost',
+    itemTapType: constants.ItemTapAction.SHARE,
+    appClassName: 'gfycat',
+    shareEvent: 'shareMattermost',
+    appId: 'mattermostwebviews',
+    enableHistory: true,
+    header: {
+        tabs: [constants.Tab.TRENDING, constants.Tab.REACTIONS],
+        displayText: false,
+    },
+};
+
+type Props = {
+    onGifClick?: (gif: string) => void;
+    defaultSearchText?: string;
+    handleSearchTextChange: (text: string) => void;
+}
+
+const GifPicker = (props: Props) => {
+    const [action, setAction] = useState(props.defaultSearchText ? 'search' : 'trending');
+
+    const handleTrending = () => setAction('trending');
+    const handleCategories = () => setAction('reactions');
+    const handleSearch = () => setAction('search');
+
+    const handleItemClick = (gif: GfycatAPIItem) => {
+        props.onGifClick?.('![' + gif.title?.replace(/]|\[/g, '\\$&') + '](' + gif.max5mbGif + ')');
```

Comment:
@sinansonmez I know you didn't add this function but would it be ok to fix the sanitisation error? We recently added this check and its been very helpful for our webapp

Reply:
I don't think this alert is actually correct. It seems to think we're doing something more risky than generating Markdown text. Worst case scenario, the user's post will be malformed rather than something worse happening

Reply:
I agree with @hmhealey. To be honest, I also didn't get which backslash character I should escape. I am not excellent at regex 😵‍💫 

I also see that conversation is marked as resolved by code-scanning-bot